app/src/main/java/com/keylesspalace/tusky/TabPreferenceActivity.kt

```diff
@@ -261,28 +275,85 @@ class TabPreferenceActivity : BaseActivity(), Injectable, ItemInteractionListene
+        statusLayout.gravity = Gravity.CENTER
+        val progress = ProgressBar(this)
+        val preferredPadding = getDimensionFromAttribute(this, androidx.appcompat.R.attr.dialogPreferredPadding)
+        progress.setPadding(preferredPadding, 0, preferredPadding, 0)
+        progress.visible(false)
+
+        val noListsText = TextView(this)
+        noListsText.setPadding(preferredPadding, 0, preferredPadding, 0)
+        noListsText.text = getText(R.string.select_list_empty)
+        noListsText.visible(false)
+
+        statusLayout.addView(progress)
+        statusLayout.addView(noListsText)
+
+        val dialogBuilder = AlertDialog.Builder(this)
+            .setTitle(R.string.select_list_title)
+            .setNeutralButton(R.string.select_list_create) { _, _ ->
+                val listIntent = Intent(applicationContext, ListsActivity::class.java)
+                startActivity(listIntent)
+            }
+            .setNegativeButton(android.R.string.cancel, null)
+            .setView(statusLayout)
+            .setAdapter(adapter) { _, position ->
+                val list = adapter.getItem(position)
+                val newTab = createTabDataFromId(LIST, listOf(list!!.id, list.title))
+                currentTabs.add(newTab)
+                currentTabsAdapter.notifyItemInserted(currentTabs.size - 1)
+                updateAvailableTabs()
+                saveTabs()
+            }
+
+        val showProgressBarJob = getProgressBarJob(progress, 500)
+        showProgressBarJob.start()
+
+        val dialog = dialogBuilder.show()
+
         lifecycleScope.launch {
             mastodonApi.getLists().fold(
                 { lists ->
+                    showProgressBarJob.cancel()
                     adapter.addAll(lists)
+                    if (lists.isNotEmpty()) {
+                        dialog.getButton(DialogInterface.BUTTON_NEUTRAL).hide()
+                        dialog.setTitle(R.string.select_list_title)
+                    } else {
+                        noListsText.show()
+                    }
                 },
                 { throwable ->
+                    dialog.hide()
                     Log.e("TabPreferenceActivity", "failed to load lists", throwable)
+                    Snackbar.make(binding.root, R.string.error_list_load, Snackbar.LENGTH_LONG).show()
                 }
             )
         }
+    }
 
-        AlertDialog.Builder(this)
-            .setTitle(R.string.select_list_title)
-            .setAdapter(adapter) { _, position ->
-                val list = adapter.getItem(position)
-                val newTab = createTabDataFromId(LIST, listOf(list!!.id, list.title))
-                currentTabs.add(newTab)
-                currentTabsAdapter.notifyItemInserted(currentTabs.size - 1)
-                updateAvailableTabs()
-                saveTabs()
-            }
-            .show()
+    private fun getDimensionFromAttribute(context: Context, attr: Int): Int {
```

Comment:
should probably be in `com.keylesspalace.tusky.util.ThemeUtils` where we have similar utils? Also it could be an extension function on `Context`

Reply:
Wait, isn't `getDimension` in `ThemeUtils` doing the same already?

Reply:
Ok, used that.

(Didn't find that as it uses completely different objects/methods.)